index.bs

```diff
@@ -2156,10 +2189,121 @@ Base64-encoded string.
 
 TODO: Full page screenshots, and multiple output formats
 
+<div algorithm>
+To <dfn>normalise rect</dfn> given |rect|:
+
+Note: This ensures that the resulting rect has positive [=width dimension=] and
+[=height dimension=].
+
+1. Let |x| be |rect|'s [=x coordinate=].
+
+1. Let |y| be |rect|'s [=y coordinate=].
+
+1. Let |width| be |rect|'s [=width dimension=].
+
+1. Let |height| be |rect|'s [=height dimension=].
+
+1. If |width| is less than 0, set |x| to |x| - |width| and then set |width| to
```

Comment:
should it be `|x| + |width|` given that the width is negative? IIUC, the idea is to flip the origin of the rect? for example, `x = 10; width = -10;` should become `x = 0; width = 10;`? unless `|width|` means taking an absolute number?

Reply:
Good catch! The `|` definitely make it seem like a module (absolute value), but in this context it is not. I believe your comment is correct.

Likewise, the same applies to the `height`.

Reply:
Indeed, good catch.